substrate/frame/revive/src/benchmarking.rs

```diff
@@ -2330,6 +2336,283 @@ mod benchmarks {
 		assert_eq!(meter.consumed(), <T as Config>::WeightInfo::v1_migration_step() * 2);
 	}
 
+	// Helper function to create a test signer for finalize_block benchmark
+	fn create_test_signer<T: Config>() -> (T::AccountId, SigningKey, H160) {
+		use hex_literal::hex;
+		// dev::alith()
+		let signer_account_id = hex!("f24FF3a9CF04c71Dbc94D0b566f7A27B94566cac");
+		let signer_priv_key =
+			hex!("5fb92d6e98884f76de468fa3f6278f8807c48bebc13595d45af5bdc4da702133");
+
+		let signer_key = SigningKey::from_bytes(&signer_priv_key.into()).expect("valid key");
+
+		let signer_address = H160::from_slice(&signer_account_id);
+		let signer_caller = T::AddressMapper::to_fallback_account_id(&signer_address);
+
+		(signer_caller, signer_key, signer_address)
+	}
+
+	// Helper function to extract contract events from system events
+	fn get_contract_events<T: Config>() -> Vec<
+		frame_system::EventRecord<
+			<T as frame_system::Config>::RuntimeEvent,
+			<T as frame_system::Config>::Hash,
+		>,
+	> {
+		frame_system::Pallet::<T>::events()
+			.into_iter()
+			.filter(|event_record| {
+				// Use string representation to identify contract events
+				format!("{:?}", event_record.event).contains("ContractEmitted")
+			})
+			.collect()
+	}
+
+	// Helper function to create and sign a transaction for finalize_block benchmark
+	fn create_signed_transaction(
+		signer_key: &SigningKey,
+		target_address: H160,
+		value: U256,
+	) -> Vec<u8> {
+		let unsigned_tx: TransactionUnsigned = TransactionLegacyUnsigned {
+			to: Some(target_address),
+			value,
+			chain_id: Some(1.into()),
+			input: vec![].into(),
+			..Default::default()
+		}
+		.into();
+
+		let hashed_payload = sp_io::hashing::keccak_256(&unsigned_tx.unsigned_payload());
+		let (signature, recovery_id) =
+			signer_key.sign_prehash_recoverable(&hashed_payload).expect("signing success");
+
+		let mut sig_bytes = [0u8; 65];
+		sig_bytes[..64].copy_from_slice(&signature.to_bytes());
+		sig_bytes[64] = recovery_id.to_byte();
+
+		let signed_tx = unsigned_tx.with_signature(sig_bytes);
+
+		signed_tx.signed_payload()
+	}
+
+	// Macro to generate common finalize_block benchmark setup
+	macro_rules! setup_finalize_block_benchmark {
```

Comment:
looks like you could use a regular function instead of a macro can't you?

Reply:
Since there is yet another benchmark `finalize_block_event_data_processing` I am leaving it as a separate function. If you decide to merge both benchmarks into one I will inline it.